perun-core/src/main/java/cz/metacentrum/perun/core/entry/GroupsManagerEntry.java

```diff
@@ -856,17 +858,30 @@ public void removeAdmin(PerunSession sess, Group group, Group authorizedGroup) t
 	}
 
 	@Override
-	public Paginated<RichGroup> getGroupsPage(PerunSession sess, Vo vo, GroupsPageQuery query, List<String> attrNames) throws VoNotExistsException, PrivilegeException {
+	public Paginated<RichGroup> getGroupsPage(PerunSession sess, Vo vo, GroupsPageQuery query, List<String> attrNames) throws VoNotExistsException, PrivilegeException, MemberNotExistsException, GroupNotExistsException, MemberGroupMismatchException {
 		Utils.checkPerunSession(sess);
 		perunBl.getVosManagerBl().checkVoExists(sess, vo);
 
-		if (!AuthzResolver.authorizedInternal(sess, "getGroupsPage_Vo_GroupsPageQuery_List<String>_policy", vo)) {
+		if (query.getMemberId() != null) {
+			perunBl.getMembersManagerBl().checkMemberExists(sess, perunBl.getMembersManagerBl().getMemberById(sess, query.getMemberId()));
+		}
+
+		// Authorization
+		if (query.getMemberId() != null) {
+			if (!AuthzResolver.authorizedInternal(sess, "member-getGroupsPage_Vo_GroupsPageQuery_List<String>_policy",  perunBl.getMembersManagerBl().getMemberById(sess, query.getMemberId()))) {
+				throw new PrivilegeException(sess, "getGroupsPage");
+			}
+		} else if (!AuthzResolver.authorizedInternal(sess, "getGroupsPage_Vo_GroupsPageQuery_List<String>_policy", vo)) {
 			throw new PrivilegeException(sess, "getGroupsPage");
 		}
 
 		Paginated<RichGroup> result = groupsManagerBl.getGroupsPage(sess, vo, query, attrNames);
 		result.setData(getGroupsManagerBl().filterOnlyAllowedAttributes(sess, result.getData(), null, true));
```

Comment:
I suggest calling this overloaded method:
```suggestion
		result.setData(getGroupsManagerBl().filterOnlyAllowedAttributes(sess, result.getData(), member, null, true));
```

Then we won't need the next if statement.